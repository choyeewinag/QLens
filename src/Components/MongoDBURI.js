import React, { useState, useEffect } from "react"; // userEffect to fetch instead of component mount
import '../public/index.css';

// input field for the mongoDB uri
const MongoDBURI = ({schemaData, uriData, submitbtn, geturi}) =>  {
    return (
        <div className="formContainer">
            <div className="formClass">
              <form className="form__group field">
                  <input className="form__field" type="input" onChange= {geturi} id='name' placeholder=" Input Your MongoDB URI"/>
                  <label for="name" className="form__label"> MongoDB URI </label>
                  <input className="URISubmitButton" type="submit" value="Submit" onClick={submitbtn} style={{display: "none"}}/>
              </form>
            </div>
        </div>
    )
}

export default MongoDBURI;
