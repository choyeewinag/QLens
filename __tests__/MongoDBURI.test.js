import React, { useState, useEffect } from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import { render, fireEvent, queryByTestId, queryByPlaceholderText } from '@testing-library/react';
import MongoDBURI from '../src/Components/MongoDBURI';
import '@testing-library/jest-dom/extend-expect'

document.createRange = () => {
  const range = new Range();

  range.getBoundingClientRect = jest.fn();

  range.getClientRects = () => {
    return {
      item: () => null,
      length: 0,
      [Symbol.iterator]: jest.fn()
    };
  };

  return range;
}

// const Button = ({onClick, submitbtn }) => (
//   <input onClick={onClick}> {submit}</input>
// )

// xtest('calls onClick prop when clicked', () => {
//   const handleClick = jest.fn()
//   render(<Button onClick={handleClick}>Click Me</Button>)
//   fireEvent.click(screen.getByTest(/click me/i))
//   expect(handleClick).toHaveBeenCalledTimes(1)
// })


//https://testing-library.com/docs/dom-testing-library/example-intro/


//Basic Test with React-test-renderer
xit('renders correctly react-test-renderer', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<MongoDBURI />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });


  describe("Input value", () => {
    xit ("updates on change", () =>{
      const {queryByPlaceholderText} = render(<MongoDBURI />)

      const URIInput = queryByPlaceholderText(' Input Your MongoDB URI');

      fireEvent.change(URIInput, {target: {value: "test"}})

      expect(URIInput.value).toBe("test")
    })

  })


//placeholdertext and ID works  
xit('MongoURI placeholder works correctly', () => {
  const {queryByPlaceholderText} = render(<MongoDBURI />)
  expect(queryByPlaceholderText(" Input Your MongoDB URI")).toBe(" Input Your MongoDB URI")
})
xit('MongoURI input box exists', () => {
  const {queryByTestId} = render(<MongoDBURI />)
  expect(queryByTestId("MongoURIInputButton")).toBe("MongoURIInputButton")
})