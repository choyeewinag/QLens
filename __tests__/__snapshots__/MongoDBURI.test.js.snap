// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly react-test-renderer 1`] = `
<div
  className="formContainer"
>
  <div
    className="formClass"
  >
    <form
      className="form__group field"
    >
      <input
        className="form__field"
        data-testId="MongoURIInputButton"
        id="name"
        placeholder=" Input Your MongoDB URI"
        type="input"
      />
      <label
        className="form__label"
        for="name"
      >
         MongoDB URI 
      </label>
      <input
        className="URISubmitButton"
        style={
          Object {
            "display": "none",
          }
        }
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;
